src/replication/include/replication/state.hpp

```diff
@@ -85,7 +86,7 @@ struct ReplicationState {
 
   auto IsMainWriteable() const -> bool {
     if (auto const *main = std::get_if<RoleMainData>(&replication_data_)) {
-      return main->writing_enabled_;
+      return !ha_cluster_ || main->writing_enabled_;
```

Comment:
I would just rename it, maybe to `part_of_ha_cluster` or something that means membership